behave/features/e2e_test_features/pipeline_validation_steps/pipeline_validation_steps.py

```diff
@@ -0,0 +1,111 @@
+from behave import given, when, then
+
+import boto3
+from datetime import timedelta, date
+import csv
+import hashlib
+import os
+
+env = os.environ['ENVIRONMENT']
+salt = os.environ['SALT']
+region = os.environ['REGION']
+
+s3 = boto3.client('s3')
+glue_client = boto3.client('glue', region_name=region)
+
+
+@given('the data is available for yesterday')
+def the_data_is_available_for_yesterday(context):
+    '''
+    This will always be true. The data is inserted in Yesterday's run since pipeline needs to cut-off data before
+    mid night yesterday.
+    '''
+    pass
+
+
+@when('the pipeline is processed')
+def run_daily_pipeline(context):
+    '''
+    This will always be true as well. Pipeline is run in staging as part of Gatling stress test.
+    '''
+    pass
+
+
+@then('nhs_number "{nhs_number}" is available in LA feed for "{hub}" for yesterday')
+def assert_nhs_number_is_available_in_hub(context, nhs_number, hub):
+    la_results = get_la_result_for_user(nhs_number, hub)
+    assert len(la_results) == 1
+
+
+@then('nhs_number "{nhs_number}" is available in supermarket feed for yesterday')
+def assert_nhs_number_is_available_in_supermarket(context, nhs_number):
+    supermarket_results = get_supermarket_result_for_user(nhs_number)
+    assert len(supermarket_results) == 1
+
+
+def get_la_result_for_user(nhs_number, hub):
+    la_results = get_latest_la_file_contents(hub)
+    print("Using NHS number: {}".format(nhs_number))
```

Comment:
nit: it might be nicer to use [f-Strings](https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python) but i don't think you should bother to change it.